appservice/src/appSettings/downloadAppSettings.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebSiteManagementModels } from "@azure/arm-appservice";
+import * as fse from 'fs-extra';
+import * as vscode from 'vscode';
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { AppSettingsTreeItem } from "../tree/AppSettingsTreeItem";
+import { confirmOverwriteSettings } from "./confirmOverwriteSettings";
+import { IAppSettingsClient } from "./IAppSettingsClient";
+import { ILocalSettingsJson } from "./ILocalSettingsJson";
+
+export async function downloadAppSettings(node: AppSettingsTreeItem, localSettingsFileName: string, localSettingsPath: string): Promise<void> {
```

Comment:
> The refactor requires you to pass in the localFileSettings name and file path since it can vary from extension to extension (specifically App Service vs. SWA and Functions).

I get that app service would be different, but wouldn't SWA and Functions be the same?

> Also, only Functions supports encryption/decryption since it relies on the func cli core tools. The refactor would require the Functions extension to do the decryption before the download and the encryption afterwards.

Again, I feel like SWA should be the same as Functions in terms of encryption/decryption

Reply:
It is, but we don't have the hard dependency on the cli tools like Functions.  Technically you can still use the app settings without having the Function extension as well.  So not really sure how to handle that scenario.

Reply:
Functions doesn't have a hard dependency on the cli tools, either. We only need the func cli if the settings are encrypted, and SWA would have the same problem